README.md

```diff
@@ -0,0 +1,230 @@
+copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the 
+Software, and to permit persons to whom the Software is furnished to do so, subject
+to the following conditions:
+
+* Redistributions of source code must retain the above copyright notice, this list 
+  of conditions and the following disclaimers.
+
+* Redistributions in binary form must reproduce the above copyright notice, this list
+  of conditions and the following disclaimers in the documentation and/or other materials
+  provided with the distribution.
+
+* Neither the names of the Innovative Systems Lab, the National Center for Supercomputing
+  Applications, nor the names of its contributors may be used to endorse or promote products
+  derived from this Software without specific prior written permission.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, 
+INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE CONTRIBUTORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT 
+OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+DEALINGS WITH THE SOFTWARE.
+
+## Compile and Run
+
+### Compile
+
+Inside the source directory, run:
+```bash
+mkdir build
+cd build
+cmake ..
+make
+```
+
+Note: In CMake, `..` is the path to the source directory.
+
+We also provide the package `cuda-memtest` in the [Spack package manager](https://spack.io) .
+
+### Run
+
+```
+cuda_memtest
+```
+The default behavior is running the test on all the GPUs available infinitely.
+There are options to change the default behavior. 
+
+```
+cuda_memtest --disable_all --enable_test 10
+cuda_memtest --stress
+```
+This runs test 10 (the stress test). `--stress` is equivalent to `--disable_all --enable_test 10 --exit_on_error`
+
+```
+cuda_memtest --stress --num_iterations 100 --num_passes 1
+```
+This one does a quick sanity check for GPUs with a short run of test 10. More on this later.
+
+See help message by 
+
+```
+cuda_memtest --help
+```
+
+### Sanity Check
+
+There is a simple script `sanity_check.sh` in the directory. 
+This script does a quick check if one GPU or all GPUs are in bad health.
+
+Example usage: 
+```bash
+# copy the cuda_memtest binary first into the same location as this script, e.g.
+cd ..
+mv build/cuda_memtest .
+```
+```
+./sanity_check.sh 0   //check GPU 0
+./sanity_check.sh 1   //check GPU 1 
+./sanity_check.sh     //check All GPUs in the system
+```
+
+Fork note: We just run the `cuda_memtest` binary directly.
+Consider this script as a source for inspiration, or so.
+
+### Known Issues
+
+* If your machine is cuda 2.2, killing the program while it is running test 10 (the memory stress test) could result 
+  in your GPUs in bad state. This is a bug from the nvidia driver. A detailed description can be found in 
```

Comment:
I think there is a word missing between `GPUs` and `in`, like `being` or `remaining`.

Reply:
`nvidia` is also written as `Nvidia`, both versions are in several places.

Reply:
Or retro NVidia? Nobody knows